autopush/endpoint.py

```diff
@@ -744,8 +744,9 @@ def post(self, uaid=""):
         self.start_time = time.time()
         self.add_header("Content-Type", "application/json")
         params = self._load_params()
+        # If the client didn't provide a CHID, make one up.
         if "channelID" not in params:
-            return self._error(400, "Invalid arguments")
+            params["channelID"] = uuid.uuid4()
```

Comment:
it would eventually get serialized to a dashed string, but I'm fine explicitly using hex. We should probably also fix the uaid fetch later to do the same so we're consistent.